clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/view/AmazonClusterProvider.groovy

```diff
@@ -203,9 +203,13 @@ class AmazonClusterProvider implements ClusterProvider<AmazonCluster>, ServerGro
       allImages
     )
 
-    Collection<AmazonCluster> clusters = cacheResults[CLUSTERS.ns].collect { clusterData ->
+    Collection<AmazonCluster> clusters = cacheResults[CLUSTERS.ns].findResults { clusterData ->
       Map<String, String> clusterKey = Keys.parse(clusterData.id)
 
+      if (clusterKey == null) {
```

Comment:
now titus clusters are also indexed by application and end up in the results. The key parser returns null for a key from a different provider so we'll filter them out in here

(the same wasn't necessary in TitusClusterProvider because it doesn't use getAllByApplication - though maybe it should...)